presto-parser/src/main/java/com/facebook/presto/sql/tree/treeutil/BinaryLiteralUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree.treeutil;
+
+import com.facebook.presto.sql.parser.ParsingException;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+/**
+ * Created by yyy on 11/17/15.
+ */
+public class
+BinaryLiteralUtil
+{
+    private BinaryLiteralUtil(){}
+
+    public static Slice fromHexVarchar(Slice slice)
```

Comment:
Use Guava's `BaseEncoding.base16()`. This code doesn't need to be super efficient (i.e., by operating directly on Slice), so no need to roll your own.